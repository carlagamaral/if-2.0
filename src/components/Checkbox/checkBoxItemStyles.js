import styled from 'styled-components';

export const Label = styled.label`
	display: flex;
`;

export const WhiteCheckSquare = styled.div`
	border: 3px solid;
	border-radius: 4px;
	display: inline-block;
	width: 35px;
	height: 35px;
	margin-right: 10px;
	z-index: 1;
	background-color: ${({ check }) => (check ? '#fff' : 'transparent')};
`;
