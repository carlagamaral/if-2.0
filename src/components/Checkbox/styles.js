import styled from 'styled-components';

export const WrapperCheckbox = styled.fieldset`
	font-size: 26px;
	border: none;
	padding: 0;
	margin: 0;
	input {
		display: none;
	}
`;
