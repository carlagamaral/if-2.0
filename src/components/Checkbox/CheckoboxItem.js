import React from 'react';
import PropTypes from 'prop-types';
import { Label, WhiteCheckSquare } from './checkBoxItemStyles';

export const CheckboxItem = (props) => {
	const { disabled, isChecked, label, onChange, testId } = props;

	return (
		<Label>
			<input
				checked={isChecked}
				data-testid={testId}
				disabled={disabled}
				onChange={onChange}
				type="checkbox"
				value={label}
			/>
			<WhiteCheckSquare check={isChecked} />
			{label}
		</Label>
	);
};

CheckboxItem.propTypes = {
	disabled: PropTypes.bool,
	isChecked: PropTypes.bool,
	label: PropTypes.string,
	onChange: PropTypes.func,
	testId: PropTypes.string,
};

CheckboxItem.defaultProps = {
	disabled: true,
	isChecked: false,
	label: '',
	onChange: () => {},
	testId: '',
};

export default CheckboxItem;
