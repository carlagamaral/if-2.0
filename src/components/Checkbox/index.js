import React from 'react';
import PropTypes from 'prop-types';
import { CheckboxItem } from './CheckoboxItem';
import { WrapperCheckbox } from './styles';

const Checkbox = ({
	click,
	disabled,
	isChecked,
	hasPadding,
	label,
	testId,
}) => {
	return (
		<WrapperCheckbox hasPadding={hasPadding}>
			<div data-testid={`div-${testId}`}>
				<CheckboxItem
					disabled={disabled}
					isChecked={isChecked}
					label={label}
					onChange={() => (disabled ? null : click())}
					testId={testId}
					value={label}
				/>
			</div>
		</WrapperCheckbox>
	);
};

Checkbox.propTypes = {
	click: PropTypes.func,
	disabled: PropTypes.bool,
	flexDirection: PropTypes.string,
	hasPadding: PropTypes.bool,
	isChecked: PropTypes.bool,
	label: PropTypes.string.isRequired,
	testId: PropTypes.string,
};

Checkbox.defaultProps = {
	click: () => {},
	disabled: false,
	flexDirection: '',
	hasPadding: false,
	isChecked: false,
	testId: '',
};

export default Checkbox;
