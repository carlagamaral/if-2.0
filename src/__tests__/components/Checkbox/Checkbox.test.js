/* eslint-disable function-paren-newline */
import 'jest-dom/extend-expect'
import { cleanup, waitForElement, render, fireEvent } from 'react-testing-library'
import React from 'react'
import Checkbox from '../../../components/Checkbox'

describe( 'Checkbox tests ', () => {
  afterEach( () => {
    cleanup()
    jest.clearAllMocks()
  } )

  const mockedFunction = jest.fn()

  test( 'Label is written in input label', async() => {
    const { container, getByTestId } = render(
      <Checkbox
        label="Naruto"
        testId="checkbox"
      />
    )
    const input = await waitForElement( () => getByTestId( 'checkbox' ) )
    const generalDiv = await waitForElement( () => getByTestId( 'div-checkbox' ) )

    expect( input.value ).toBe( 'Naruto' )
    expect( generalDiv ).toBeInTheDocument()
    expect( container ).toMatchSnapshot()
  } )

  test( 'When there is no prop "isChecked" the component is not checked by default', async() => {
    const { container, getByTestId } = render(
      <Checkbox
        label="Naruto"
        testId="checkbox"
      />
    )
    const input = await waitForElement( () => getByTestId( 'checkbox' ) )
    const generalDiv = await waitForElement( () => getByTestId( 'div-checkbox' ) )

    expect( input.checked ).toBe( false )
    expect( generalDiv ).toBeInTheDocument()
    expect( container ).toMatchSnapshot()
  } )

  test( 'When there is a prop "isChecked" with value False the component is not checked', async() => {
    const { container, getByTestId } = render(
      <Checkbox
        isChecked={ false }
        label="Naruto"
        testId="checkbox"
      />
    )
    const input = await waitForElement( () => getByTestId( 'checkbox' ) )
    const generalDiv = await waitForElement( () => getByTestId( 'div-checkbox' ) )

    expect( input.checked ).toBe( false )
    expect( generalDiv ).toBeInTheDocument()
    expect( container ).toMatchSnapshot()
  } )

  test( 'When there is a prop "isChecked" with value True the component is checked', async() => {
    const { container, getByTestId } = render(
      <Checkbox
        isChecked={ true }
        label="Naruto"
        testId="checkbox"
      />
    )
    const input = await waitForElement( () => getByTestId( 'checkbox' ) )
    const generalDiv = await waitForElement( () => getByTestId( 'div-checkbox' ) )

    expect( input.checked ).toBe( true )
    expect( generalDiv ).toBeInTheDocument()
    expect( container ).toMatchSnapshot()
  } )

  test( 'When there is no prop "disabled" the component is abled by default', async() => {
    const { container, getByTestId } = render(
      <Checkbox
        label="Naruto"
        testId="checkbox"
      />
    )
    const input = await waitForElement( () => getByTestId( 'checkbox' ) )
    const generalDiv = await waitForElement( () => getByTestId( 'div-checkbox' ) )

    expect( input.disabled ).toBe( false )
    expect( generalDiv ).toBeInTheDocument()
    expect( container ).toMatchSnapshot()
  } )

  test( 'When there is a prop "disabled" with value True the component is disabled', async() => {
    const { container, getByTestId } = render(
      <Checkbox
        disabled={ true }
        label="Naruto"
        testId="checkbox"
      />
    )
    const input = await waitForElement( () => getByTestId( 'checkbox' ) )
    const generalDiv = await waitForElement( () => getByTestId( 'div-checkbox' ) )

    expect( input.disabled ).toBe( true )
    expect( generalDiv ).toBeInTheDocument()
    expect( container ).toMatchSnapshot()
  } )

  test( 'When there is a prop "disabled" with value False the component is abled', async() => {
    const { container, getByTestId } = render(
      <Checkbox
        disabled={ false }
        label="Naruto"
        testId="checkbox"
      />
    )
    const input = await waitForElement( () => getByTestId( 'checkbox' ) )
    const generalDiv = await waitForElement( () => getByTestId( 'div-checkbox' ) )

    expect( input.disabled ).toBe( false )
    expect( generalDiv ).toBeInTheDocument()
    expect( container ).toMatchSnapshot()
  } )

  test( 'When there is a "click" prop, the function is called when the input is clicked', async() => {
    const { container, getByTestId } = render(
      <Checkbox
        click={ mockedFunction }
        label="Naruto"
        testId="checkbox"
      />
    )
    const input = await waitForElement( () => getByTestId( 'checkbox' ) )
    const generalDiv = await waitForElement( () => getByTestId( 'div-checkbox' ) )

    fireEvent.click( input )

    expect( mockedFunction ).toHaveBeenCalled()
    expect( generalDiv ).toBeInTheDocument()
    expect( container ).toMatchSnapshot()
  } )

  test( 'When there is a "click" prop and a true "disabled" prop, the function is not called when the input is clicked', async() => {
    const { container, getByTestId } = render(
      <Checkbox
        click={ mockedFunction }
        disabled={ true }
        label="Naruto"
        testId="checkbox"
      />
    )
    const input = await waitForElement( () => getByTestId( 'checkbox' ) )
    const generalDiv = await waitForElement( () => getByTestId( 'div-checkbox' ) )

    fireEvent.click( input )

    expect( mockedFunction ).not.toHaveBeenCalled()
    expect( generalDiv ).toBeInTheDocument()
    expect( container ).toMatchSnapshot()
  } )
} )
