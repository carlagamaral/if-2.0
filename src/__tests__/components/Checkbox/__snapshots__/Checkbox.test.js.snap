// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox tests  Label is written in input label 1`] = `
<div>
  <fieldset
    class="sc-AxjAm iQWkcR"
  >
    <div
      class="checkbox "
      data-testid="div-checkbox"
    >
      <label>
        <input
          data-testid="checkbox"
          type="checkbox"
          value="Naruto"
        />
        <span
          class="checkbox-material"
        >
          <span
            class="check"
          />
        </span>
        Naruto
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`Checkbox tests  When there is a "click" prop and a true "disabled" prop, the function is not called when the input is clicked 1`] = `
<div>
  <fieldset
    class="sc-AxjAm iQWkcR"
  >
    <div
      class="checkbox "
      data-testid="div-checkbox"
    >
      <label>
        <input
          data-testid="checkbox"
          disabled=""
          type="checkbox"
          value="Naruto"
        />
        <span
          class="checkbox-material"
        >
          <span
            class="check"
          />
        </span>
        Naruto
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`Checkbox tests  When there is a "click" prop, the function is called when the input is clicked 1`] = `
<div>
  <fieldset
    class="sc-AxjAm iQWkcR"
  >
    <div
      class="checkbox "
      data-testid="div-checkbox"
    >
      <label>
        <input
          data-testid="checkbox"
          type="checkbox"
          value="Naruto"
        />
        <span
          class="checkbox-material"
        >
          <span
            class="check"
          />
        </span>
        Naruto
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`Checkbox tests  When there is a prop "disabled" with value False the component is abled 1`] = `
<div>
  <fieldset
    class="sc-AxjAm iQWkcR"
  >
    <div
      class="checkbox "
      data-testid="div-checkbox"
    >
      <label>
        <input
          data-testid="checkbox"
          type="checkbox"
          value="Naruto"
        />
        <span
          class="checkbox-material"
        >
          <span
            class="check"
          />
        </span>
        Naruto
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`Checkbox tests  When there is a prop "disabled" with value True the component is disabled 1`] = `
<div>
  <fieldset
    class="sc-AxjAm iQWkcR"
  >
    <div
      class="checkbox "
      data-testid="div-checkbox"
    >
      <label>
        <input
          data-testid="checkbox"
          disabled=""
          type="checkbox"
          value="Naruto"
        />
        <span
          class="checkbox-material"
        >
          <span
            class="check"
          />
        </span>
        Naruto
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`Checkbox tests  When there is a prop "isChecked" with value False the component is not checked 1`] = `
<div>
  <fieldset
    class="sc-AxjAm iQWkcR"
  >
    <div
      class="checkbox "
      data-testid="div-checkbox"
    >
      <label>
        <input
          data-testid="checkbox"
          type="checkbox"
          value="Naruto"
        />
        <span
          class="checkbox-material"
        >
          <span
            class="check"
          />
        </span>
        Naruto
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`Checkbox tests  When there is a prop "isChecked" with value True the component is checked 1`] = `
<div>
  <fieldset
    class="sc-AxjAm iQWkcR"
  >
    <div
      class="checkbox "
      data-testid="div-checkbox"
    >
      <label>
        <input
          checked=""
          data-testid="checkbox"
          type="checkbox"
          value="Naruto"
        />
        <span
          class="checkbox-material"
        >
          <span
            class="check"
          />
        </span>
        Naruto
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`Checkbox tests  When there is no prop "disabled" the component is abled by default 1`] = `
<div>
  <fieldset
    class="sc-AxjAm iQWkcR"
  >
    <div
      class="checkbox "
      data-testid="div-checkbox"
    >
      <label>
        <input
          data-testid="checkbox"
          type="checkbox"
          value="Naruto"
        />
        <span
          class="checkbox-material"
        >
          <span
            class="check"
          />
        </span>
        Naruto
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`Checkbox tests  When there is no prop "isChecked" the component is not checked by default 1`] = `
<div>
  <fieldset
    class="sc-AxjAm iQWkcR"
  >
    <div
      class="checkbox "
      data-testid="div-checkbox"
    >
      <label>
        <input
          data-testid="checkbox"
          type="checkbox"
          value="Naruto"
        />
        <span
          class="checkbox-material"
        >
          <span
            class="check"
          />
        </span>
        Naruto
      </label>
    </div>
  </fieldset>
</div>
`;
